src/query_parser/v3/lexer.rl

```diff
@@ -346,44 +366,235 @@ main := |*
       fbreak;
     }
   };
+
+    wildcard_tag => {
+    tok.numval = 0;
+    tok.len = 1;
+    tok.s = ts;
+    tok.pos = tok.s - q->raw;
+    RSQuery_Parse_v3(pParser, LB, tok, q);
+    if (!QPCTX_ISOK(q)) {
+      fbreak;
+    }
+
+    int is_attr = (*(ts + 3) == '$') ? 1 : 0;
+    tok.type = is_attr ? QT_PARAM_WILDCARD : QT_WILDCARD;
+    tok.len = te - (ts + 3 + is_attr) - 2;
+    tok.s = ts + 3 + is_attr;
+    tok.pos = tok.s - q->raw;
+    RSQuery_Parse_v3(pParser, WILDCARD, tok, q);
+    if (!QPCTX_ISOK(q)) {
+      fbreak;
+    }
+
+    tok.len = 1;
+    tok.s = te - 1;
+    tok.pos = tok.s - q->raw;
+    RSQuery_Parse_v3(pParser, RB, tok, q);
+    if (!QPCTX_ISOK(q)) {
+      fbreak;
+    }
+  };
+
+  unescaped_tag => {
+    tok.numval = 0;
+    tok.len = 1;
+    tok.s = ts;
+    tok.pos = tok.s - q->raw;
+    RSQuery_Parse_v3(pParser, LB, tok, q);
+    if (!QPCTX_ISOK(q)) {
+      fbreak;
+    }
+
+    tok.len = te - (ts + 2);
+    tok.s = ts + 1;
+
+    // remove leading spaces
```

Comment:
No need to handle escaping for leading/trailing spaces since they are trimmed upon ingestion and should also be trimmed when querying?

Reply:
Yes, leading/trailing spaces are trimmed upon ingestion:
```sh
127.0.0.1:6379> FT.CREATE idx schema t tag
OK
127.0.0.1:6379> hset doc1 t '   foo bar'
(integer) 1
127.0.0.1:6379> hset doc2 t 'foo bar'
(integer) 1
127.0.0.1:6379> FT.TAGVALS idx t
1) "foo bar"
127.0.0.1:6379> hset doc3 t 'foo bar    '
(integer) 1
127.0.0.1:6379> FT.TAGVALS idx t
1) "foo bar"
```